spec/unit/pops/time/timespan_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'spec_helper'
+require 'puppet/pops'
+require 'puppet_spec/compiler'
+
+module Puppet::Pops
+module Time
+describe 'Timespan' do
+  include PuppetSpec::Compiler
+
+  let! (:simple) { Timespan.from_fields(false, 1, 3, 10, 11) }
+  let! (:all_fields_hash) { {'days' => 1, 'hours' => 7, 'minutes' => 10, 'seconds' => 11, 'milli_seconds' => 123, 'micro_seconds' => 456, 'nano_seconds' => 789} }
+  let! (:complex) { Timespan.from_fields_hash(all_fields_hash) }
+
+  context 'can be created from a String' do
+    it 'using default format' do
+      expect(Timespan.parse('1-03:10:11')).to eql(simple)
+    end
+
+    it 'using explicit format' do
+      expect(Timespan.parse('1-7:10:11.123456789', '%D-%H:%M:%S.%N')).to eql(complex)
+    end
+
+    it 'using leading minus and explicit format' do
+      expect(Timespan.parse('-1-7:10:11.123456789', '%D-%H:%M:%S.%N')).to eql(-complex)
+    end
+
+    it 'using %H as the biggest quantity' do
+      expect(Timespan.parse('27:10:11', '%H:%M:%S')).to eql(simple)
+    end
+
+    it 'using %M as the biggest quantity' do
+      expect(Timespan.parse('1630:11', '%M:%S')).to eql(simple)
+    end
+
+    it 'using %S as the biggest quantity' do
+      expect(Timespan.parse('97811', '%S')).to eql(simple)
+    end
+
+    it 'using %L as the biggest quantity' do
+      expect(Timespan.parse('97811000', '%L')).to eql(simple)
+    end
+
+    it 'using %N as the biggest quantity' do
+      expect(Timespan.parse('112211123456789', '%N')).to eql(complex)
+    end
+
+    it 'where biggest quantity is not frist' do
+      expect(Timespan.parse('11:1630', '%S:%M')).to eql(simple)
+    end
+  end
```

Comment:
Yes, you missed it. Check p_timespan_type_spec.rb, line 63 and p_timestamp_type_spec.rb line 64.